src/core/eval.ml

```diff
@@ -331,27 +343,32 @@ and snf : term -> term = fun t ->
   | Wild        -> assert false
   | TRef(_)     -> assert false
 
+(** [snf ctx t] computes the strong normal form of term [t] in context
+    [ctx]. *)
+let snf : ctxt -> term -> term = fun ctx t -> snf (Ctxt.to_llet ctx t)
+
 (** [whnf t] computes a weak head-normal form of [t]. *)
-let whnf : term -> term = fun t ->
+let whnf : ctxt -> term -> term = fun ctx t ->
   Stdlib.(steps := 0);
   let t = unfold t in
-  let u = whnf t in
+  let u = whnf (Ctxt.to_llet ctx t) in
```

Comment:
Finally, you decided to remove the ctx argument from whnf_stk. Why? Was it more efficient?

Reply:
It was similar on holide, and slightly more efficient on the fibonacci test